fix(TRSNPC): Updated to .SetupUpText + feat: Added Shadow Veil support

Baconadors opened the pull request:
Updated the deprecated TRSNPC.Setup to .SetupUpText and added support for casting Shadow Veil from the Arceuus spellbook.

Torwent commented:
Heya! Thanks for the PR. I might wait until I go and return from vacation to accept it since I have to look into what you did on the blast furnace one. As for ardy knights, all looks good!

Baconadors commented:
> Heya! Thanks for the PR. I might wait until I go and return from vacation to accept it since I have to look into what you did on the blast furnace one. As for ardy knights, all looks good!

So here's what I did on blast furnace:

1. I added support for Smith's gloves (i) as when you get the normal Smith's gloves from giant's foundry, you can combine them with the ice gloves to create Smith's gloves (i). To make it simple, I added a check box in the GUI that would change the string from 'Ice gloves' (default glvoes when new box is unchecked) to 'Smith's gloves (i)'.

2. I change the location of the check boxes from horizontal to vertical, keeping all functionality the same.
https://i.imgur.com/VUkENK8.png

3. There have been some complaints in help-chat that the script would get stuck in a loop where it was continuously withdrawing and depositing coal/ores. Essentially this guys complaint: https://discord.com/channels/795071177475227709/1099786910253203638/1229543830731358259
So the check checks the current bar so it knows whether you have just an ore or ore & coal. The first check is to see if you have a stack of 28 ore in the bank, and if the stack is <= 28, the script terminates because there isn't enough ore to fill a whole inventory.
There's a secondary check if you're making a bar that needs coal, and does the same check with the same criteria of needing more than 28 coal, or the script terminates.

4. I also noticed when testing gold that the script was not withdrawing the goldsmith gauntlets when the script started. Lines 707 & 710 make sure you have either the goldsmith gauntlets or a coal bag, depending on what bar is selected.

Still learning, so hopefully this is up to your standards. If not, feel free to make any modifications when you're home form vacation.
If I see any other issues brought up on either of these scripts, I may post the changes in #mods until this can get committed.
